packages/api/models/granules.js

```diff
@@ -67,40 +66,41 @@ class Granule extends Manager {
    * @returns {Promise<undefined>} - undefined
    */
   async reingest(g) {
-    await this.applyWorkflow(g, 'IngestGranule', 'input');
+    const { name, version } = deconstructCollectionId(g.collectionId);
+
+    // get the payload of the original execution
+    const status = await aws.StepFunction.getExecutionStatus(path.basename(g.execution));
+    const originalMessage = JSON.parse(status.execution.input);
+
+    const lambdaPayload = await Rule.buildPayload({
+      workflow: 'IngestGranule',
+      meta: originalMessage.meta,
+      payload: originalMessage.payload,
+      provider: g.provider,
+      collection: {
+        name,
+        version
+      }
+    });
+
+    await this.updateStatus({ granuleId: g.granuleId }, 'running');
+
+    await aws.invoke(process.env.invoke, lambdaPayload);
   }
 
   /**
    * apply a workflow to a given granule object
    *
    * @param {Object} g - the granule object
    * @param {string} workflow - the workflow name
-   * @param {string} messageSource - 'input' or 'output' from previous execution
-   * @param {Object} metaOverride - overrides the meta of the new execution,
-   *                                accepts partial override
-   * @param {Object} payloadOverride - overrides the payload of the new execution,
-   *                                   accepts partial override
    * @returns {Promise<undefined>} undefined
    */
-  async applyWorkflow(g, workflow, messageSource, metaOverride, payloadOverride) {
+  async applyWorkflow(g, workflow) {
     const { name, version } = deconstructCollectionId(g.collectionId);
 
-    // get the payload of the original execution
-    const status = await aws.StepFunction.getExecutionStatus(path.basename(g.execution));
-    const originalMessage = JSON.parse(status.execution[messageSource]);
-
-    const meta = metaOverride
-      ? merge(originalMessage.meta, metaOverride)
-      : originalMessage.meta;
-
-    const workflowPayload = payloadOverride
-      ? merge(originalMessage.payload, payloadOverride)
-      : originalMessage.payload;
-
     const lambdaPayload = await Rule.buildPayload({
       workflow,
-      meta,
-      workflowPayload,
+      payload: g,
```

Comment:
Should it be `granules []`?  IngestGranule and tasks takes `granules` array as input payload.

Reply:
IngestGranule should still go to reingest, but you're right that it would make it easier to work with the tasks to pass a `{ granules: [ g ] }` payload instead. I'll make that change.